### ISSUE TITLE
Run keeps prematurely ending...

### ISSUE THREAD

**Issue by rmormando** [rrwick/Polypolish#14]
Hello I am trying to use the filter alignments and Polypolish step but for both commands the run doesn't fully finish.

For the filter alignments I run this command:

```
polypolish_insert_filter.py --in1 11_align_1.sam --in2 11_align_2.sam --out1 11_filtered_1.sam --out2 11_filtered_2.sam
```

And I get this error message at the end:

```
Loading alignments (2023-04-27 18:11:31)
11_align_1.sam: 137,423,308 alignments from 1,392,225 reads
11_align_2.sam: Killed
```

And there is nothing outputted after that at all.

The same thing happens when I go to run the Polypolish command:

```
polypolish 11.fasta 11_align_1.sam 11_align_2.sam > 11_polished.fasta
```

I get all the reads and their lengths output on the STDOUT but then it says `Killed` and then it's just done.

I'm not sure what's wrong with it - I've tried running it on Bioconda and from the source but I don't get a finished run.

**Comment by kowrithv** [comment]
I have exactly the same problem. Have you found a solution for the problem in the meantime.

**Comment by rrwick** [comment]
I suspect the error here is a lack of memory. In the original post, there were a _lot_ of alignments: 137 million from only 1.4 million reads, which makes me think the genome is very repetitive.

You can check this by running it through `time`, e.g. `/usr/bin/time -v polypolish_insert_filter.py ...` to see its max memory usage. If it does indeed hit your system's limits, then the only solution is running on a computer with more RAM :cry:

Ryan

**Comment by rrwick** [comment]
I have just released [Polypolish v0.6.0](https://github.com/rrwick/Polypolish/releases/tag/v0.6.0), in which the insert-size filtering has been implemented in Rust. This should be more resource-efficient than the previous Python implementation, so runs that previously ran out of memory may work now 🤞
